tests/unit/workspace_access/test_verification.py

```diff
@@ -79,3 +79,114 @@ def test_fail_get_permissions_missing_object(caplog):
     manager.verify_applied_permissions("ot1", "oid1", state, "backup")
     manager.verify_roles_and_entitlements(state, "backup")
     assert caplog.messages == ['removed on backend: ot1.oid1', 'removed on backend: wid1']
+
+
+def test_verify(mocker):
+    ws = create_autospec(WorkspaceClient)
+    secret_scopes = SecretScopesSupport(ws)
+    manager = VerificationManager(ws, secret_scopes)
+
+    state = MigrationState(
+        [
+            MigratedGroup(
+                id_in_workspace="wid1",
+                name_in_workspace="test",
+                name_in_account="test",
+                temporary_name="db-temp-test",
+                members=None,
+                entitlements=None,
+                external_id="eid1",
+                roles=None,
+            )
+        ]
+    )
+    verify_applied_scope_acls = mocker.patch(
+        "databricks.labs.ucx.workspace_access.verification.VerificationManager.verify_applied_scope_acls"
```

Comment:
Don't monkey patch any "databricks.*" code. This test is not easy to change and would be missed by static code analysis, when we change internals of the class.

inject mocks of dependencies through constructor instead and verify that dependencies are called.

Reply:
I have removed all patches and applied MagicMock and dependency injection wherever possible.